### PR TITLE
Samples table enhancements followup

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR is a follow up to #679.

* Added in action cable config and updated the Projects Table so that it is now refreshed when a Sample is created, updated, or deleted.
* Moved hidden input handling from `infinite-scroll` controller into a new `form--hidden-inputs` controller.
* Updated `clone` and `transfer` samples response to reload the page
* Added `data-turbo-permanent` to `dialog` and `flash` so that if the page is refreshed they persist.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Launch server
2. Navigate to Project Samples page for Project of choice
3. Perform actions on the table that require a selection of samples and ensure that each respond as expected

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/javascript/controllers/form/hidden_inputs_controller.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { createHiddenInput } from "utilities/form";
+
+export default class extends Controller {
+    static outlets = ["selection"];
+    static values = {
+        fieldName: String,
+    };
+
+    connect() {
+        this.allIds = this.selectionOutlet.getStoredItems();
```
Comment (joshsadam) [inline app/javascript/controllers/form/hidden_inputs_controller.js:11]: **nitpick:** I know this has nothing to do with this PR, but can you update this?  There is no reason to keep all the stored items as an attribute on this controller, it is a waste of storage.  Maybe pass it as a variable or something into appendHiddenInputs.

Comment (ericenns) [reply app/javascript/controllers/form/hidden_inputs_controller.js]: Fixed in [f64be2d](https://github.com/phac-nml/irida-next/pull/691/commits/f64be2d1ad8c8b297f1ae7e37a06c696c193484a)

**Review by ChrisHuynh333 — APPROVED** [review]
Looks great! Nothing to add on my end

**Review by joshsadam — COMMENTED** [review]
Everything seems to be working, I just asked for one thing.

**Review by joshsadam — APPROVED** [review]

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Pull request merged**